src/swap/useSwapQuote.ts

```diff
@@ -20,88 +21,83 @@ const useSwapQuote = () => {
   const walletAddress = useSelector(walletAddressSelector)
   const useGuaranteedPrice = useSelector(guaranteedSwapPriceEnabledSelector)
   const [exchangeRate, setExchangeRate] = useState<ExchangeRate | null>(null)
-  const [fetchSwapQuoteError, setFetchSwapQuoteError] = useState(false)
-  const [fetchingSwapQuote, setFetchingSwapQuote] = useState(false)
-
-  useEffect(() => {
-    setFetchingSwapQuote(false)
-  }, [exchangeRate])
 
   // refreshQuote requests are generated when the swap input amounts are
   // changed, but the quote response / updated exchange rate updates the swap
   // input amounts. this variable prevents duplicated requests in this scenario
   const requestUrlRef = useRef<string>('')
 
-  const refreshQuote = async (
-    fromToken: TokenBalance,
-    toToken: TokenBalance,
-    swapAmount: ParsedSwapAmount,
-    updatedField: Field
-  ) => {
-    setFetchSwapQuoteError(false)
+  const refreshQuote = useAsyncCallback(
+    async (
+      fromToken: TokenBalance,
+      toToken: TokenBalance,
+      swapAmount: ParsedSwapAmount,
+      updatedField: Field
+    ) => {
+      if (!swapAmount[updatedField].gt(0)) {
+        setExchangeRate(null)
```

Comment:
setters should be done in `onSuccess` or `onError`, so they don't fire if the component has been unmounted.